dd-sdk-android-core/src/main/kotlin/com/datadog/android/core/internal/metrics/BatchMetricsDispatcher.kt

```diff
@@ -89,7 +90,7 @@ internal class BatchMetricsDispatcher(
         deletionReason: RemovalReason
     ): Map<String, Any?>? {
         val fileCreationTimestamp = file.nameAsTimestampSafe(internalLogger) ?: return null
-        val fileAgeInMillis = dateTimeProvider.getDeviceTimestamp() - fileCreationTimestamp
+        val fileAgeInMillis = max(0, (dateTimeProvider.getDeviceTimestamp() - fileCreationTimestamp))
```

Comment:
Then it would still skew, but towards the zero. maybe it is better to ignore than this metric if it is negative?
Or maybe we don't need to use `dateTimeProvider` here? Because file is created using `newFileName = System.currentTimeMillis().toString()`, which is not affected at all by NTP synchronization. Meaning that to calculate file age property, we still need to use `System.currentTimeMillis()` directly, this will be reliable, because we want duration, and not instant.

wdyt?

Reply:
iOS has the same and we wanted to align Android also to make sure we have the same charts. I am ready to discuss this into the parking lot today during our daily.

Reply:
The thing is that file created using `System.currentTimeMillis()`, without NTP-adjustment (well, maybe there is adjustment done by the device). It means that for file age at least we need also to use `System.currentTimeMillis()` without NTP-adjustment, this __should__ guarantee that time behind these calls is monotonic.

Reply:
exactly why we are using that here and not the NTP adjustment one. But no one can guarantee that by the time we send the metric the System time was not skewed or the time zone was not changed. This will give a negative age. We've already noticed this behaviour in our data.